src/util/compat.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+export opaque type Compatible<T> = [Compat, T];
+type Compat = {|
+  +type: string,
+  +version: string,
+|};
+
+export function toCompat<T>(
+  type: string,
+  version: string,
+  obj: T
+): Compatible<T> {
+  return [{type, version}, obj];
+}
+
+/**
+ * Load an object from compatibilized state created by `toCompat`.
+ * The object has an expected type and version, and may optionally have
+ * handler functions for transforming previous versions into a canonical state.
+ * If a handler is present for the current version, it will be applied.
+ * Throws an error if the compatibilized object is the wrong type, or if its version
+ * is not current and there was no handler for its version.
+ */
+export function fromCompat<T>(
+  expectedType: string,
+  expectedVersion: string,
+  obj: Compatible<T>,
```

Comment:
This really is not a `Compatible<T>`; it is a `Compatible<mixed>`.

If we wanted to be more precise, we could write something using `$ObjMap` on `handlers` below, but using `any`/`mixed` here I think is acceptable, because appropriate types would just be unified away, anyway.

Reply:
Changed to any